runtime/browser/runtime_download_manager_delegate.h

```diff
@@ -40,6 +51,33 @@ class RuntimeDownloadManagerDelegate
  private:
   friend class base::RefCountedThreadSafe<RuntimeDownloadManagerDelegate>;
 
+  struct DownloadSelectFileParams {
+    DownloadSelectFileParams(content::DownloadItem* item,
+                             const content::DownloadTargetCallback& callback);
+    ~DownloadSelectFileParams();
+
+    content::DownloadItem* item;
+    content::DownloadTargetCallback callback;
+  };
+
+#if defined(OS_LINUX) && !defined(OS_TIZEN)
+  class RuntimeDownloadView : public content::WebContentsObserver {
```

Comment:
DownloadManagerDelegate is not aware of UI and it should not. pls take a look at 'xwalk/runtime/browser/runtime_ui_delegate_desktop.h', the 'RuntimeUIDelegateDesktop' class is the place where UI should be "attached" to a Runtime instance.